arrow/parser.py

```diff
@@ -339,6 +507,7 @@ def _parse_token(self, token, value, parts):
             parts["year"] = 1900 + value if value > 68 else 2000 + value
 
         elif token in ["MMMM", "MMM"]:
+            # FIXME: month_number() is nullable
             parts["month"] = self.locale.month_number(value.lower())
```

Comment:
We should refactor this.
`month_number()` is nullable

Reply:
Would you like to take on this refactor?

Reply:
@jadchaar When I change type of `parser._Parts.month` from `int` to `Optional[int]`, mypy prints

```
arrow/parser.py:656: error: Argument 2 to "datetime" has incompatible type "Optional[int]"; expected "int"
arrow/parser.py:697: error: Argument "month" to "datetime" has incompatible type "Optional[int]"; expected "int"
```

But I don't have any clue to fix this because I'm not familiar with `_build_datetime()` and `parts` is mutable. Can you help with this?

Reply:
@isac322 maybe we can refactor this after the PR is merged?